django/contrib/auth/management/__init__.py

```diff
@@ -62,36 +69,33 @@ def create_permissions(
     try:
         app_config = apps.get_app_config(app_label)
         ContentType = apps.get_model("contenttypes", "ContentType")
-        Permission = apps.get_model("auth", "Permission")
     except LookupError:
         return
 
     if not router.allow_migrate_model(using, Permission):
         return
 
+    models = list(app_config.get_models())
+    # Grab all the content types from the current database.
+    ctypes = ContentType.objects.db_manager(using).get_for_models(
+        *models, for_concrete_models=False
+    )
     # This will hold the permissions we're looking for as
     # (content_type, (codename, name))
     searched_perms = []
-    # The codenames and ctypes that should exist.
-    ctypes = set()
-    for klass in app_config.get_models():
-        # Force looking up the content types in the current database
-        # before creating foreign keys to them.
-        ctype = ContentType.objects.db_manager(using).get_for_model(
-            klass, for_concrete_model=False
+    for model in models:
+        ctype = ctypes[model]
+        searched_perms.extend(
+            [(ctype, perm) for perm in _get_all_permissions(model._meta)]
```

Comment:
We iterate over `searched_perms`, can we use a generator instead? :thinking: 
```
searched_perms = (
    [(ctypes[model], perm) for perm in _get_all_permissions(model._meta)]
    for model in models
)
```

Reply:
We can actually get rid of `searched_perms` altogether and just do the looping below.